### PR TITLE
improve subscription types

### PR DESCRIPTION
**Pull request by PabloSzx**
- The included `pubsub.subscribe` actually returns a Node.js Readable (technically from `readable-stream`, but both are pretty much the same types)
- `withFilter` types were missing

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -17,9 +17,10 @@ import {
 } from "graphql";
 import { SocketStream } from "fastify-websocket"
 import { IncomingMessage, IncomingHttpHeaders, OutgoingHttpHeaders } from "http";
+import { Readable } from "stream";
 
 export interface PubSub {
-  subscribe<TResult = any>(topics: string | string[]): Promise<AsyncIterator<TResult>>;
+  subscribe(topics: string | string[]): Promise<Readable>;
```
Comment (mcollina) [inline index.d.ts:23]: Most people does not know that a Readable is an asyncIterator :/.

**Review by mcollina — COMMENTED** [review]
+1, just a note.

File: index.d.ts (hunk 1 above)
Comment (PabloSzx) [reply index.d.ts:23]: I was playing with the types and it seems like `Readable & AsyncIterable<TResult>` seems to be a good middle-ground, while allowing to access the readable functions & keeping the async iterable stuff with the data type

https://codesandbox.io/s/summer-butterfly-xgxdv?file=/src/index.ts

Comment (mcollina) [reply index.d.ts:23]: Agree!

**Comment by PabloSzx** [comment]
I had to add TypeScript as dev dep because of `Error: Failed to load parser '@typescript-eslint/parser' declared in 'CLIOptions': Cannot find module 'typescript'` was showing up 🤷‍♂️ 

And to test async iterable stuff properly, the target should be at least set `es2018`

**Review by mcollina — APPROVED** [review]
lgtm

File: tsconfig.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "target": "es6",
+    "target": "es2018",
```
Comment (mcollina) [inline tsconfig.json:3]: why this change?

Comment (PabloSzx) [reply tsconfig.json:3]: The async iterable stuff comes with typescript es2018, without it the tests won't recognize it's keywords
